lib/load-tasks.js

```diff
@@ -1,18 +1,30 @@
 'use strict';
 const RunTask = require('runtask');
 const async = require('async');
+const cachedTasks = {};
+
 module.exports = function(config, log, done) {
   const runner = new RunTask();
-
-  async.mapValues(config.tasks, (task, name, next) => {
+  // get list of all tasks:
+  const taskList = Object.keys(config.tasks);
+  // register all tasks with a wrapper that will lazy-load/cache it if its actually used
+  async.each(taskList, (name, next) => {
+    const task = config.tasks[name];
     let taskFn;
     if (Array.isArray(task)) { //task maps to other tasks
       taskFn = task;
-    } else {
-      const Cls = require(task);
-      //if task is config, pass the whole thing, little bit of a hack
-      const taskConfig = (config[name] && config[name].needsEntireConfig) ? config : config[name];
-      taskFn = new Cls(name, taskConfig, runner, log);
+    } else { // task is wrapped by a function that will lazy-load/cache it:
+      taskFn = (callback) => {
+        if (cachedTasks[name]) {
```

Comment:
do a if !cachedTask and then set the value and then always do cachedTask[].execute(callback).